internal/server/singleprocess/state/state.go

```diff
@@ -121,6 +132,39 @@ func (s *State) Close() error {
 	return s.db.Close()
 }
 
+// Prune should be called in a on a regular interval to allow State
+// to prune out old data.
+func (s *State) Prune() error {
+	memTxn := s.inmem.Txn(true)
+	defer memTxn.Abort()
+
+	jobs, err := s.jobsPruneOld(memTxn, maximumJobsIndexed)
+	if err != nil {
+		return err
+	}
+
+	var records int
+
+	for _, f := range pruneFns {
+		cnt, err := f(memTxn)
+		if err != nil {
+			return err
+		}
+
+		records += cnt
+	}
+
+	s.log.Debug("Finished pruning data",
+		"removed-jobs", jobs,
+		"removed-records", records,
+		"op-tables", len(pruneFns),
+	)
```

Comment:
Might be useful to dump this per-function maybe as a trace so we can see prunes by table.

Reply:
That's also reported now, per table.